src/Entity/SmsMessage.php

```diff
@@ -444,7 +448,11 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
     $fields['recipient_entity'] = BaseFieldDefinition::create('dynamic_entity_reference')
       ->setLabel(t('Recipient entity'))
       ->setDescription(t('The entity who received the SMS message.'))
-      ->setRequired(FALSE);
+      ->setRequired(FALSE)
+      ->setSettings([
+        'exclude_entity_types' => TRUE,
```

Comment:
I've not gone through the code in detail, but it looks to me like DER already provides those defaults. This should imply a bug in DER.
```
class DynamicEntityReferenceItem extends EntityReferenceItem {

  /**
   * {@inheritdoc}
   */
  public static function defaultStorageSettings() {
    return [
      'exclude_entity_types' => TRUE,
      'entity_type_ids' => [],
    ];
  }
```